hazelcast/src/main/java/com/hazelcast/util/executor/StripedExecutor.java

```diff
@@ -164,59 +177,64 @@ private Worker getWorker(Runnable command) {
 
     final class Worker extends Thread {
 
-        private final BlockingQueue<Runnable> workQueue;
+        private final BlockingQueue<Runnable> taskQueue;
         private final SwCounter processed = SwCounter.newSwCounter();
         private final int queueCapacity;
+        private final AtomicBoolean started = new AtomicBoolean();
 
         private Worker(String threadNamePrefix, int queueCapacity) {
             super(threadNamePrefix + "-" + THREAD_ID_GENERATOR.incrementAndGet());
-            this.workQueue = new LinkedBlockingQueue<Runnable>(queueCapacity);
+            this.taskQueue = new LinkedBlockingQueue<Runnable>(queueCapacity);
             this.queueCapacity = queueCapacity;
         }
 
-        private void schedule(Runnable command) {
-            long timeout = 0;
-            TimeUnit timeUnit = TimeUnit.SECONDS;
-            if (command instanceof TimeoutRunnable) {
-                TimeoutRunnable timeoutRunnable = ((TimeoutRunnable) command);
-                timeout = timeoutRunnable.getTimeout();
-                timeUnit = timeoutRunnable.getTimeUnit();
+        private void schedule(Runnable task) {
+            if (!started.get() && started.compareAndSet(false, true)) {
```

Comment:
`!started.get()` is superfluous here

Reply:
From a functional point of view yes, from a performance point of view not. You do not want to have a cas for every put because this would require a lot more cache coherence traffic.

Reply:
I needed to confirm this as well.

Below is a JMH benchmark:

```
package com;


import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.Mode;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.Setup;
import org.openjdk.jmh.annotations.State;
import org.openjdk.jmh.annotations.Threads;
import org.openjdk.jmh.annotations.Warmup;
import org.openjdk.jmh.infra.Blackhole;

import java.util.concurrent.TimeUnit;
import java.util.concurrent.atomic.AtomicBoolean;


@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.NANOSECONDS)
@State(Scope.Benchmark)
@Fork(value = 1, warmups = 1)
@Warmup(iterations = 2)
@Measurement(iterations = 5)
public class CasBenchmark {

    private final AtomicBoolean b = new AtomicBoolean();

    @Benchmark
    @Threads(4)
    public boolean ccas() {
        return !b.get() && b.compareAndSet(false, true);
    }

    @Benchmark
    @Threads(4)
    public boolean cas() {
        return b.compareAndSet(false, true);
    }
}
```
And the results:

```
Benchmark          Mode  Cnt    Score   Error  Units
CasBenchmark.cas   avgt    5  115.894 ± 3.583  ns/op
CasBenchmark.ccas  avgt    5    4.350 ± 0.030  ns/op
```

So doing doing the additional read makes a big impact.

Reply:
great, thanks for sharing!